scripts/linux/smokeTestHelper.sh

```diff
@@ -0,0 +1,220 @@
+    echo "[FAIL] Mismatch Version: The latest $pkgName package version on linux repo is $1 (expecting: $2)"
+    exit 1
+  fi
+  echo "[PASS] $pkgName linux repository artifact version check ($1)"
+  echo ""
+}
+
+check-file-size-limit()
+{
+  # Compare if the incoming file size is within $sizePercentThrehold percent of
+  # the golden package standard. That is 
+  #  goldenPkgSize*(100-sizePercentThrehold)/100 < incomingPkgSize < goldenPkgSize*(100+sizePercentThrehold)/100
+  #
+  # $1 - incoming artifact
+  # $2 - golden standard artifact
+  # $3 - acceptable size difference percentage as a whole integer number
+
+  incomingArtifact=$1
+  goldenArtifact=$2
+
+  # Check if the is between 0 <= sizePercentThrehold <= 100
+  [[ "100" -lt "$3" ]] && { echo "[FAIL] The \$sizePercentThrehold ($3) needs to be lesser than 100"; exit 1; }
+  [[ "0" -gt "$3" ]] && { echo "[FAIL] The \$sizePercentThrehold ($3) needs to be greater than 0"; exit 1; }
+
+  # Do the actual size checking
+  incomingSize=$(stat -c%s $incomingArtifact)
+  goldenSize=$(stat -c%s $goldenArtifact)
+  maxAcceptingArtifactSize=$(echo "scale=0; ($goldenSize*$sizePercentThrehold/100)+$goldenSize" | bc)
+  [[ "$incomingSize" -lt "$maxAcceptingArtifactSize" ]] || { echo "[FAIL] The current artifact size ($incomingSize) is more than $edgelet.maxPercentAllowed% larger than previous version file ($goldenSize)"; exit 1; }
+  minAcceptingArtifactSize=$(echo "scale=0; $goldenSize-($goldenSize*$sizePercentThrehold/100)" | bc)
+  [[ "$incomingSize" -gt "$minAcceptingArtifactSize" ]] || { echo "[FAIL] The current artifact size ($incomingSize) is more than $edgelet.maxPercentAllowed% smaller than previous version file ($goldenSize)"; exit 1; }
+  echo "[PASS] The current artifact size ($incomingSize) is withing the range of $minAcceptingArtifactSize Byte to $maxAcceptingArtifactSize Byte" 
+}
+
+check-version-content-difference()
+{
+  # Compare if the content of two files are difference. 
+  #   Return true - if the two files are different
+  #   Return false - if the two files are byte-exact identical.
+  #
+  # $1 - path to file1
+  # $2 - path to file2
+
+  # Compare the previous and current artifact content diff (by byte)
+  if cmp --silent $1 $2; then
+    echo "[FAIL] The current ($1) and previous ($2) artifacts are identical."
+    exit 1;
+  fi
+  echo "[PASS] The current and previous artifact version have a different content"
+}
+
+check-github-pmc-artifacts-similarity()
+{
+  # Compare the artifact if github and pmc are the same artifact
+  #  Return true - if the artifacts from two sources are identical
+  #  Return false - if the artifacts from two sources are NOT identical
+  #
+  # $1 - path to PMC artifact
+  # $2 - path to Github artifact directory
+  # $3 - package name
+
+  echo ""
+  ghArtifact="$2/$3_$gitHubArtifactSuffix"
+  echo "The artifact from GitHub: $ghArtifact"
+  [[ -f "$ghArtifact" ]] || { echo "[FAIL] The artifact from GitHub doesn't exist ($ghArtifact)"; exit 1; }
```

Comment:
Should this check also be for PMC?

Reply:
Nah, PMC artifact is downloaded in the same task. The code won't get here if that fails. While GitHub artifact is downloaded in a prior job, which *may* get here if the task fails.